src/Oscar.jl

```diff
@@ -33,57 +33,80 @@ const jll_deps = String["Antic_jll", "Arb_jll", "Calcium_jll", "FLINT_jll", "GAP
                         "libpolymake_julia_jll", "libsingular_julia_jll", "msolve_jll",
                         "polymake_jll", "Singular_jll"];
 
-function _lookup_git_branch(dir::AbstractString)
-   if length(Sys.which("git")) != nothing &&
-         isdir(joinpath(dir,".git"))
+function _lookup_commit_from_cache(url::AbstractString, tree::AbstractString)
+   if Sys.which("git") != nothing
       try
-         ref = cd(dir) do
-            readchomp(`git rev-parse --abbrev-ref HEAD`)
+         path = Pkg.Types.add_repo_cache_path(url)
+         if isdir(path)
+            commit = readchomp(`sh -c "git -C $path log --oneline --all --pretty='tree %T;%H' | grep \"^tree $tree\" | cut -d\; -f2 | head -n1"`)
```

Comment:
So what's wrong with something like this:
```suggestion
            commit = readchomp(`git -C $path rev-parse HEAD`)
```
(Of course if that work, you might as well remove this line and change the next one to use `HEAD` instead of `$commit`)

However, given that the code used `git rev-parse --abbrev-ref HEAD` before, I assume there is a problem with using `HEAD`, but what is it? Is `HEAD` simply missing? I only have two dirs in `~/.julia/clones` but they both have a `HEAD`